packages/connectors/src/notification.ts

```diff
@@ -0,0 +1,115 @@
+import { SendMessageCommand } from "@aws-sdk/client-sqs";
+import { PutCommand } from "@aws-sdk/lib-dynamodb";
+import { randomUUID } from "crypto";
+import { dynamoDBDocumentClient, getAwsSQSQueueURL, sqsClient } from "./utils";
+
+const DYNAMODB_NOTIFICATION_TABLE_NAME = "Notification";
+
+/**
+ * Creates a notification record in DynamoDB and enqueues it for immediate sending.
+ *
+ * @param notificationId - Unique identifier for the notification
+ * @param emails - Array of email addresses to send the notification to
+ * @param subject - Email subject line
+ * @param body - Email body content
+ */
+const sendImmediate = async ({
+  notificationId,
+  emails,
+  subject,
+  body,
+}: {
+  notificationId?: string;
+  emails: string[];
+  subject: string;
+  body: string;
+}): Promise<void> => {
+  const id = notificationId ?? randomUUID();
+  try {
+    await _createNotificationRecord({ notificationId: id, emails, subject, body });
+    await _enqueueDeferredNotification(id!);
+  } catch (error) {
+    throw `Error creating immediate notification id ${id} - ${(error as Error).message}`;
```

Comment:
We should add the `cause` here in case we need the stack

i.e. 

throw new Error("Connecting to database failed.", { cause: err });

Also looks like your throwing a string?